pkg/controller/statefulset/stateful_set.go

```diff
@@ -78,13 +78,15 @@ type StatefulSetController struct {
 
 // NewStatefulSetController creates a new statefulset controller.
 func NewStatefulSetController(
+	ctx context.Context,
 	podInformer coreinformers.PodInformer,
 	setInformer appsinformers.StatefulSetInformer,
 	pvcInformer coreinformers.PersistentVolumeClaimInformer,
 	revInformer appsinformers.ControllerRevisionInformer,
 	kubeClient clientset.Interface,
 ) *StatefulSetController {
 	eventBroadcaster := record.NewBroadcaster()
+	logger := klog.LoggerWithName(klog.FromContext(ctx), "StatefulSet")
```

Comment:
Let's set the name in `cmd/kube-controller-manager/app/core.go` instead. When this code is called in a unit tests we don't need the extra name because there is only one controller.

Reply:
Also note that lowercase "statefuleset" might be the preferred name (@pravarag used it below and I think I have seen something similar elsewhere).